ReSharper.FSharp/src/FSharp.Psi/src/Impl/DeclaredElement/CompilerGenerated/FSharpUnionTagsClass.cs

```diff
@@ -44,6 +45,18 @@ public class FSharpUnionTagsClass : FSharpGeneratedMemberBase, IClass
     public INamespace GetContainingNamespace() =>
       Union.GetContainingNamespace();
 
+    public bool HasMemberWithName(string shortName, bool ignoreCase)
+    {
+      var comparisonRule = ignoreCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
+
+      foreach (var name in MemberNames)
```

Comment:
What do you think about caching the names in a HashSet for more efficient searches?

Reply:
What do other changed implementations do?

Reply:
Differently. For example, MemberOwner does not cache and collects names on the fly. Others use SlimHashSet.

Reply:
This class is only rarely used in resolve from C# code, so I'm more or less OK with keeping it like this for now (though, I'd personally remove the extra braces).
If doing anything better, I'd add a `SlimHashSet` to `IUnionPart`/`UnionPartBase`, so this generated class could access it.